src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala

```diff
@@ -44,23 +45,28 @@ import scala.jdk.CollectionConverters._
 import scala.util.control.NonFatal
 
 object Boot extends IOApp with LazyLogging {
-  def run(args: List[String]): IO[ExitCode] =
+
+  def run(args: List[String]): IO[ExitCode] = {
+    initSentry()
+    implicit val system = ActorSystem("sam")
+
     (startup() *> ExitCode.Success.pure[IO]).recoverWith { case NonFatal(t) =>
       logger.error("sam failed to start, trying again in 5s", t)
+
+      Http().shutdownAllConnectionPools() *> system.terminate()
+      CollectorRegistry.defaultRegistry.clear()
```

Comment:
I noticed when sam restarts locally the liveness server still hangs around and causes confusing errors. Also I noticed similar errors relating to prometheus that went away when I added L57.